crates/operator/src/controller.rs

```diff
@@ -455,6 +469,76 @@ impl K8sContext {
 
         Ok(())
     }
+
+    pub async fn deploy_node(&self) -> anyhow::Result<HydraDoomNode> {
+        info!("Deploying new node.");
+
+        // List available snapshots.
+        // Try move from available to used dir.
+        // If successful, start new node.
+        // If anything fails, at any point, deploy offline node.
+
+        let api: Api<HydraDoomNode> = Api::default_namespaced(self.client.clone());
+        let name = format!(
+            "{}{}{}",
+            self.config.autoscaler_region_prefix,
+            "0", // 1 for online, 0 for offline
+            random_name()
+        );
+        let new_node = HydraDoomNode {
+            spec: HydraDoomNodeSpec::default(),
+            status: None,
+            metadata: kube::api::ObjectMeta {
+                name: Some(name.clone()),
+                ..Default::default()
+            },
+        };
+        // Create or patch the deployment
+        api.patch(
+            &name,
+            &PatchParams::apply("hydra-doom-pod-controller"),
+            &Patch::Apply(&new_node),
+        )
+        .await
+        .map_err(|err| {
+            error!(err = err.to_string(), "Failed to create new node.");
+            err.into()
+        })
+    }
+
+    pub async fn remove_node(&self, crd: &HydraDoomNode) -> anyhow::Result<()> {
+        info!("Removing node: {}", crd.name_any());
+        todo!()
+    }
+
+    pub async fn scale(&self) -> anyhow::Result<()> {
+        let api: Api<HydraDoomNode> = Api::default_namespaced(self.client.clone());
+        let crds = api.list(&ListParams::default()).await?;
+
+        let mut available_hydra_nodes: Vec<HydraDoomNode> = crds
+            .into_iter()
+            .filter(|crd| match &crd.status {
+                Some(status) => status.game_state == String::from(HydraDoomGameState::Waiting),
+                None => false,
+            })
+            .collect();
+
+        if available_hydra_nodes.len() < self.config.autoscaler_low_watermark {
+            let amount = available_hydra_nodes.len() - self.config.autoscaler_low_watermark;
+            // One after the other to avoid race conditions.
+            for _ in 0..amount {
```

Comment:
lets put a `min(amount, MAX_SCALE_UP_BATCH)` so that jumps in new pods are batched. I'm worried that the fade-in of the operator could try to put 500 nodes at all once and get banned by fargate.